From da41d1d01e2c20838667b8b26991434f79c95d74 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Mon, 22 Apr 2019 19:10:05 +0200
Subject: [PATCH 52/75] cpuidle: annotate
 cpuidle_{pause_and_lock,resume_and_unlock}

HANDLES:

drivers/cpuidle/cpuidle.c:386:1: warning: mutex 'cpuidle_lock' is still held at the end of function [-Wthread-safety-analysis]
}
^
drivers/cpuidle/cpuidle.c:384:2: note: mutex acquired here
        mutex_lock(&cpuidle_lock);
        ^
drivers/cpuidle/cpuidle.c:396:2: warning: releasing mutex 'cpuidle_lock' that was not held [-Wthread-safety-analysis]
        mutex_unlock(&cpuidle_lock);
        ^
---
 drivers/cpuidle/cpuidle.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/cpuidle/cpuidle.c b/drivers/cpuidle/cpuidle.c
index 7f108309e871..f2f7beb99038 100644
--- a/drivers/cpuidle/cpuidle.c
+++ b/drivers/cpuidle/cpuidle.c
@@ -379,7 +379,7 @@ void cpuidle_uninstall_idle_handler(void)
 /**
  * cpuidle_pause_and_lock - temporarily disables CPUIDLE
  */
-void cpuidle_pause_and_lock(void)
+void cpuidle_pause_and_lock(void) __attribute__((acquire_capability(cpuidle_lock)))
 {
 	mutex_lock(&cpuidle_lock);
 	cpuidle_uninstall_idle_handler();
@@ -390,7 +390,7 @@ EXPORT_SYMBOL_GPL(cpuidle_pause_and_lock);
 /**
  * cpuidle_resume_and_unlock - resumes CPUIDLE operation
  */
-void cpuidle_resume_and_unlock(void)
+void cpuidle_resume_and_unlock(void) __attribute__((release_capability(cpuidle_lock)))
 {
 	cpuidle_install_idle_handler();
 	mutex_unlock(&cpuidle_lock);
-- 
2.17.1

